jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -254,6 +287,35 @@ def _jwst2data(file_obj, ext, data_label):
 
     return data, new_data_label
 
+# ---- Functions that handle input from Roman ASDF files -----
+
+
+def _roman_2d_asdf_to_glue_data(file_obj, data_label, ext=None):
+
+    if ext == '*':
+        ext_list = ['data', 'dq', 'err', 'var_poisson', 'var_rnoise']
+    elif isinstance(ext, list):
+        ext_list = ext
+    else:
+        ext_list = [ext]
+
+    for ext in ext_list:
+        comp_label = ext.lower()
+        new_data_label = f'{data_label}[{comp_label}]'
+        data = Data(label=new_data_label)
+
+        # For now, set bunit to a blank string.
+        bunit = ''
+        component = Component.autotyped(np.array(getattr(file_obj, ext)), units=bunit)
+        data.add_component(component=component, label=comp_label)
+        meta = getattr(file_obj, 'meta')
+        data.coords = getattr(meta, 'wcs')
+        data.meta.update(dict(meta))
+
+        yield data, new_data_label
+
+# ---- Functions that handle input from non-JWST and non-Roman files -----
```

Comment:
Do you mean to add this new empty "section" or did you mean to replace the existing section below?

Reply:
Fixed 👍🏻